sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -137,14 +181,14 @@ private static Mono<RequestBody> toOkHttpRequestBody(Flux<ByteBuffer> bbFlux, Ht
     private static Mono<ByteString> toByteString(Flux<ByteBuffer> bbFlux) {
         Objects.requireNonNull(bbFlux, "'bbFlux' cannot be null.");
         return Mono.using(okio.Buffer::new,
-            buffer -> bbFlux.reduce(buffer, (b, byteBuffer) -> {
-                try {
-                    b.write(byteBuffer);
-                    return b;
-                } catch (IOException ioe) {
-                    throw Exceptions.propagate(ioe);
-                }
-            }).map(b -> ByteString.of(b.readByteArray())), okio.Buffer::clear)
+                buffer -> bbFlux.reduce(buffer, (b, byteBuffer) -> {
+                    try {
+                        b.write(byteBuffer);
+                        return b;
+                    } catch (IOException ioe) {
+                        throw Exceptions.propagate(ioe);
+                    }
+                }).map(b -> ByteString.of(b.readByteArray())), okio.Buffer::clear)
```

Comment:
Does `readByteArray` perform as duplication of the `Buffer`s content? If it does is there another way this could be performed that uses a reference of the `Buffer`s body?

Reply:
It does copy bytes from pooled memory to byte array. If we were to aggregate ourselves we'd need to do similar thing i.e. allocate final byte array.

There's one optimization that can be done - using okioBuffer as base for ByteString but it's sketchy, we'd need to manage lifecycle, i.e. release buffer after request is sent.

This PR doesn't change this code, so I suggest we take this as follow up.

Reply:
Is there a particular reason here we don't want to directly use `b.readByteString()` but `ByteString.of(b.readByteArray())` ?.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/28946